docs/03-Development-Process.md

```diff
@@ -12,13 +12,75 @@ examples :
 - feature/issue-15469
 - bugfix/issue-15469
 
+Origin vs Upstream in a Forked GitHub Workflow:
+
+| Term         | What it means                                               | Typical URL example                                | Used for                                         |
+| ------------ | ----------------------------------------------------------- | -------------------------------------------------- | ------------------------------------------------ |
+| **origin**   | Your fork of the repository (remote on your GitHub account) | `https://github.com/your-username/together.git`    | Pushing your rebased changes, creating PRs       |
+| **upstream** | The original repository you forked from                     | `https://github.com/Together-100Devs/together.git` | Pulling the latest changes from the main project |
+
 Developers should follow the following steps :
 
-- Updated the `main` : `git pull origin main`
-  - It is generally advised to keep your branch consistently up to date with main
-- create feature branch : `git checkout -b feature/issue-15469`
+- Create a feature branch : `git checkout -b feature/issue-15469`
 - Develop the specified requirements
-- commit and Push the code
+- Commit locally:
+
+  - stage your chages: `git add .`
+  - commit your changes: `git commit -m "your commit message"`
+
+- It is generally advised to keep your branch consistently up to date with `main`.
+
+  - Grab the latest upstream version of main, in case main had changes: `git fetch upstream`
+
+  - upstream = grabbing all upstream branches (upstream/main, upstream/add-dependabot, ect)
+
+- Rebase your feature branch on top of the latest upstream/main. We're telling git to use the "main" branch from upstream as the new "base" for your feature branch `git rebase upstream/main`
+
+- To keep the commit history clean, we require that you squash your commits into a single commit using interactive rebase:
+
+  - This shows the list of commits since you branched off main, from oldest to newest `git log upstream/main..HEAD --reverse --oneline`
```

Comment:
For the life of me I'm struggling to find those comments, these are the closest I could find?
<img width="803" height="354" alt="image" src="https://github.com/user-attachments/assets/14bb2d6b-901f-4927-97be-567f05daa24e" />
<img width="806" height="355" alt="image" src="https://github.com/user-attachments/assets/35398cf4-5e5c-4619-8a53-bd1af37fa3db" />
But If I'm understanding the change "rebase upstream/main to git rebase -i ${their last commit}^" correctly is the interactive rebase method I changed here what you're looking for?

`git log upstream/main..HEAD --reverse --oneline`
`git rebase -i ${your-commit-SHA}^`

Reply:
> I usually just do `git log --oneline` but perhaps this is helpful for people?

I'm down with switching it with git log --oneline

From what I saw:

`git log --oneline` : since it shows all the commits in the branch history, after a rebase, the commits from main look like a continuation of the branch

`git log upstream/main..HEAD --oneline` : shows only the new commits that aren't in main yet

I went with the second because there seemed to be less risk of newbies accidentally squashing commits from main but it is way more wordy. Then again most newbies wouldn't merge commits with names they don't recognize, so the shorter git log --oneline might be best?